### PR TITLE
feat: Add Appinspect API action

### PR DESCRIPTION
**Pull request by mbruzda-splunk**
Tested here: 
https://github.com/splunk/splunk-add-on-for-google-workspace/actions/runs/5555742121/jobs/10185146305
https://github.com/splunk/splunk-add-on-for-cisco-meraki/actions/runs/5576114853

### CODE REVIEW

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -783,6 +789,39 @@ jobs:
...
+    needs: build
+    if: |
+      always() &&
+      needs.build.result == 'success'
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        tags:
+          - "cloud"
```
Comment (mkolasinski-splunk) [inline .github/workflows/reusable-build-test-release.yml:803]: What is the reasoning behind for choosing those 3 tags?

File: .github/workflows/reusable-build-test-release.yml [hunk 2]
```diff
@@ -783,6 +789,39 @@ jobs:
...
+      - uses: actions/download-artifact@v3
+        with:
+          name: package-splunkbase
+          path: build/package
+      - name: AppInspect API
+        uses: splunk/appinspect-api-action@v3.0
+        with:
+          username: ${{ secrets.SPL_COM_USER }}
+          password: ${{ secrets.SPL_COM_PASSWORD }}
+          app_path: build/package/
```
Comment (mkolasinski-splunk) [inline .github/workflows/reusable-build-test-release.yml:817]: Here you can use output of download-artifact action: https://github.com/actions/download-artifact#download-path-output

**Comment by mkolasinski-splunk** [comment]
Generally LGTM - if possible please improve logging when waiting until appinspect process the request.

File: .github/workflows/reusable-build-test-release.yml (hunk 1 above)
Comment (mbruzda-splunk) [reply .github/workflows/reusable-build-test-release.yml:803]: This is what we did during QAC

**Comment by mkolasinski-splunk** [comment]
@mbruzda-splunk  could you please update README accordingly?

File: .github/workflows/reusable-build-test-release.yml [hunk 3]
```diff
@@ -783,6 +789,39 @@ jobs:
           path: |
             *_markdown.txt
 
+  appinspect-api:
+    name: appinspect api ${{ matrix.tags }}
+    needs: build
+    if: |
```
Comment (artemrys) [inline .github/workflows/reusable-build-test-release.yml:795]: @mkolasinski-splunk do we want to run it always? or maybe only on pr and push to main?

Comment (mkolasinski-splunk) [reply .github/workflows/reusable-build-test-release.yml:795]: @artemrys yes, this is a good point.

Comment (mbruzda-splunk) [reply .github/workflows/reusable-build-test-release.yml:795]: changed that

**Comment by mbruzda-splunk** [comment]
PR to develop (appinspect api skipped):
https://github.com/splunk/splunk-add-on-for-microsoft-sysmon/actions/runs/5782636638/job/15669950299

**Comment by mbruzda-splunk** [comment]
PR to main (appinspect api job runs):
https://github.com/splunk/splunk-add-on-for-microsoft-sysmon/actions/runs/5782724104

**Review by mkolasinski-splunk — APPROVED** [review]

**Pull request merged**